From 46b7e66aa10ce1d28c92b711056d170d08bfbbf7 Mon Sep 17 00:00:00 2001
From: Chad Versace <chadversary@chromium.org>
Date: Wed, 19 Jul 2017 10:55:38 -0700
Subject: [PATCH] Update list of KHR extensions for
 dEQP-VK.api.info.*.extensions

Fixes failures on drivers that expose extensions newer than dEQP.

v2 (Bas Nieuwenhuizen): Add
                          VK_KHR_image_format_list
                          VK_KHR_bind_memory2
                          VK_KHR_relaxed_block_layout
                          VK_KHR_maintenance2
---
 .../modules/vulkan/api/vktApiFeatureInfo.cpp       | 28 ++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/external/vulkancts/modules/vulkan/api/vktApiFeatureInfo.cpp b/external/vulkancts/modules/vulkan/api/vktApiFeatureInfo.cpp
index 8b98f082..ddf2dded 100644
--- a/external/vulkancts/modules/vulkan/api/vktApiFeatureInfo.cpp
+++ b/external/vulkancts/modules/vulkan/api/vktApiFeatureInfo.cpp
@@ -656,6 +656,11 @@ void checkInstanceExtensions (tcu::ResultCollector& results, const vector<string
 		"VK_KHR_xcb_surface",
 		"VK_KHR_xlib_surface",
 		"VK_KHR_get_physical_device_properties2",
+		"VK_KHR_get_surface_capabilities2",
+		"VK_KHR_external_memory_capabilities",
+		"VK_KHR_external_semaphore_capabilities",
+		"VK_KHR_external_fence_capabilities",
+
 	};
 
 	checkKhrExtensions(results, extensions, DE_LENGTH_OF_ARRAY(s_allowedInstanceKhrExtensions), s_allowedInstanceKhrExtensions);
@@ -671,6 +676,29 @@ void checkDeviceExtensions (tcu::ResultCollector& results, const vector<string>&
 		"VK_KHR_sampler_mirror_clamp_to_edge",
 		"VK_KHR_shader_draw_parameters",
 		"VK_KHR_maintenance1",
+		"VK_KHR_maintenance2",
+		"VK_KHR_push_descriptor",
+		"VK_KHR_descriptor_update_template",
+		"VK_KHR_incremental_present",
+		"VK_KHR_shared_presentable_image",
+		"VK_KHR_storage_buffer_storage_class",
+		"VK_KHR_16bit_storage",
+		"VK_KHR_get_memory_requirements2",
+		"VK_KHR_external_memory",
+		"VK_KHR_external_memory_fd",
+		"VK_KHR_external_memory_win32",
+		"VK_KHR_external_semaphore",
+		"VK_KHR_external_semaphore_fd",
+		"VK_KHR_external_semaphore_win32",
+		"VK_KHR_external_fence",
+		"VK_KHR_external_fence_fd",
+		"VK_KHR_external_fence_win32",
+		"VK_KHR_win32_keyed_mutex",
+		"VK_KHR_dedicated_allocation",
+		"VK_KHR_variable_pointers",
+		"VK_KHR_bind_memory2",
+		"VK_KHR_image_format_list",
+		"VK_KHR_relaxed_block_layout",
 	};
 
 	checkKhrExtensions(results, extensions, DE_LENGTH_OF_ARRAY(s_allowedDeviceKhrExtensions), s_allowedDeviceKhrExtensions);
-- 
2.15.1.620.gb9897f4670-goog

